pkg/cache/cache.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package cache
+
+import (
+	"database/sql"
+	"sync"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/pkg/errors"
+)
+
+const (
+	driverName = "sqlite3"
+)
+
+// Storage implements a SQLite3 caching backend for Colly.
+type Storage struct {
+	// Sqlite filename.
+	Filename string
+	// Duration till which a link is skipped.
+	Validity time.Duration
+	// Clear cache at start if true.
+	ClearCache bool
+	// Database handle.
+	dbHandle *sql.DB
+	// Mutex used for clearing cache database.
+	mu sync.RWMutex
+}
+
+// Init initializes cache database.
+func (s *Storage) Init() error {
+	// Check if db exists.
+	if s.dbHandle == nil {
```

Comment:
I'm wondering, if the database exists, do we need to go through the other steps below? Could we simply check if the database and table exists (optionally, truncate the table if `s.ClearCache` true) and if yes, exist, otherwise create new table? Would that flow make sense?

Reply:
Let me try that out and get back! 🙂